e2e/e2e_context_test.go

```diff
@@ -438,15 +439,17 @@ func TestE2EContext_PerRequestRetry(t *testing.T) {
 		ctx, cancel := context.WithCancel(context.Background())
 		defer cancel()
 
-		e := WithConfig(Config{
-			BaseURL:          ts.URL,
-			AssertionHandler: &mockAssertionHandler{},
+		e := httpexpect.WithConfig(httpexpect.Config{
+			BaseURL: ts.URL,
+			AssertionHandler: &httpexpect.DefaultAssertionHandler{
+				Formatter: &httpexpect.DefaultFormatter{},
+			},
 		})
```

Comment:
Good catch, actually using mockAssertionHandler here looks like a bug because it will ignore failures.

Thought, new usage is not quite correct too: DefaultAssertionHandler requires Reporter field to be non-nil. Otherwise instead of reporting failure it will panic. I pushed a commit to catch this kind of misuses earlier: 67c5e0aa9eff1214259c54080aa63e5e2baf3d38

We could set Reporter field to `t`, but even better instead of setting `Config.AssertionHandler` we can just set `Config.Reporter` to `t`. In this case httpexpect will automatically use DefaultAssertionHandler with proper fields.

So it would be:

```
		e := httpexpect.WithConfig(httpexpect.Config{
			BaseURL: ts.URL,
			Reporter: t,
		})
```

Reply:
Thank you for the detailed feedback - incorporated your comments. Will work on the remainder files.

Will also check if there is a way to get a formatter or linter to order the imports in the way we want.

Reply:
> Will also check if there is a way to get a formatter or linter to order the imports in the way we want.

Oh, it would be perfect.

Reply:
BTW welcome to [discord chat](https://discord.gg/5SCPCuCWA9) if you'll want to discuss something :)